### ISSUE TITLE
Incompatible with React v16

### ISSUE THREAD

**Issue by atschaef** [Wufe/react-particles-js#13]
Upgrading to the latest version of React v16 causes type conflicts within react-particles-js.  

If the react-particles-js dependencies are updated to the latest versions of `react`, `react-dom`, `@types/react` , & `@types/react-dom`, then all type errors are resolved, however a runtime error occurs when attempting to render particles:
<img width="582" alt="screen shot 2017-09-28 at 9 08 29 am" src="https://user-images.githubusercontent.com/6707250/30974484-02d7dbb2-a42d-11e7-9501-32144ad7650b.png">

I'm guessing this is due to changes in react-dom in v16.

**Comment by ashbhat** [comment]
Having the same issue with v16

**Comment by Gitii** [comment]
It is not a bug with react-dom v16 but with react v16.
It turned out that this package modifies a props-object directly. React v16 seems to make the props-object read only.
I fixed this bug (-> deep clone before modification using lodash.deepClone) and provided a fork.
Updated all packages and added react as a peerDependency.
https://github.com/Gitii/react-particles-js
`npm install https://github.com/Gitii/react-particles-js/tarball/master`
